gluten-core/src/main/scala/org/apache/spark/sql/metric/SQLColumnarShuffleMetricsReporter.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.metric
+
+import org.apache.spark.SparkContext
+import org.apache.spark.executor.TempShuffleReadMetrics
+import org.apache.spark.sql.execution.metric.{SQLMetric, SQLMetrics, SQLShuffleReadMetricsReporter}
+
+class SQLColumnarShuffleReadMetricsReporter(
+    tempMetrics: TempShuffleReadMetrics,
+    metrics: Map[String, SQLMetric])
+  extends SQLShuffleReadMetricsReporter(tempMetrics, metrics) {
+
+  private[this] val _batchesRead =
```

Comment:
Could you add some documents here to clarify the difference between `_batchesRead` and `_recordsRead`, and why we need them?

Reply:
I added some comments, could you please take a look again?